static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/ArchiveEntry.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+class ArchiveEntry {
+
+  private static final ArchiveEntry NOT_CLASS =
+      new ArchiveEntry("", /* isClass= */ false, /* shouldBeSkipped= */ true);
+
+  private final String className;
+  private final String classPath;
+  private final boolean isClass;
+  private final boolean shouldBeSkipped;
+
+  private ArchiveEntry(String className, boolean isClass, boolean shouldBeSkipped) {
+    this.className = className;
+    this.classPath = className.replace(".", "/");
+    this.isClass = isClass;
+    this.shouldBeSkipped = shouldBeSkipped;
+  }
+
+  static ArchiveEntry ofFilePath(String path) {
```

Comment:
```suggestion
  static ArchiveEntry fromPath(String path) {
```

Reply:
Changed to "fromZipEntryName" to reflect what's expected as an input.